cmd/ztest/ztest.c

```diff
@@ -5397,6 +5400,57 @@ ztest_spa_rename(ztest_ds_t *zd, uint64_t id)
 	(void) rw_unlock(&ztest_name_lock);
 }
 
+void
+ztest_fletcher(ztest_ds_t *zd, uint64_t id)
+{
+	hrtime_t end = gethrtime() + NANOSEC;
+
+	srandom(time(NULL));
+
+	fletcher_4_init();
+	while (gethrtime() <= end) {
+		int run_count = 100;
+		const uint32_t maxsize = 16 * 1024 * 1024;
+		void *buf;
+		uint32_t size;
+		int *ptr;
+		int i;
+		zio_cksum_t zc_ref;
+		zio_cksum_t zc_ref_byteswap;
+
+		/* 512 bytes aligned */
+		size = (random() | 0x10000) % maxsize & ~511;
```

Comment:
I'd recommend using `ztest_random_blocksize()` function for this with `umem_zalloc(., UMEM_NOFAIL);  Something like:

```
        blocksize = ztest_random_blocksize();
        buf = umem_alloc(blocksize, UMEM_NOFAIL);
```